watcher/tool/watcher/main.go

```diff
@@ -17,46 +17,101 @@ limitations under the License.
+		"INFLUXDB_GRAFANA_USERNAME":  "influxdb-grafana-username",
+		"INFLUXDB_GRAFANA_PASSWORD":  "influxdb-grafana-password",
+		"INFLUXDB_TELEGRAF_USERNAME": "influxdb-telegraf-username",
+		"INFLUXDB_TELEGRAF_PASSWORD": "influxdb-telegraf-password",
+	}
+
+	rootCmd = &cobra.Command{
+		Use:   "watcher",
+		Short: "Utility to manage InfluxDB/Grafana/Alerts",
+		RunE:  root,
+	}
+)
+
+func init() {
+	level := logrus.InfoLevel
+
+	rootCmd.PersistentFlags().StringVar(&mode, "mode", "", fmt.Sprintf("Watcher mode: %v", constants.AllModes))
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBAdminUser, "influxdb-admin-username", constants.InfluxDBAdminUser, "InfluxDB administrator username")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBAdminPassword, "influxdb-admin-password", constants.InfluxDBAdminUser, "InfluxDB administrator password")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBGrafanaUser, "influxdb-grafana-username", constants.InfluxDBGrafanaUser, "InfluxDB grafana username")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBGrafanaPassword, "influxdb-grafana-password", constants.InfluxDBGrafanaUser, "InfluxDB grafana password")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBTelegrafUser, "influxdb-telegraf-username", constants.InfluxDBTelegrafUser, "InfluxDB telegraf username")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBTelegrafPassword, "influxdb-telegraf-password", constants.InfluxDBTelegrafUser, "InfluxDB telegraf password")
+	rootCmd.PersistentFlags().BoolVar(&debug, "debug", false, "Debugging mode")
+
+	trace.SetDebug(debug)
+	if debug {
+		level = logrus.DebugLevel
+	}
+	logrus.SetFormatter(&trace.TextFormatter{})
+	logrus.SetLevel(level)
+
+	bindFlagEnv(rootCmd.PersistentFlags())
+}
+
 func main() {
-	var mode string
-	flag.StringVar(&mode, "mode", "", fmt.Sprintf("watcher mode: %v", constants.AllModes))
-	flag.Parse()
+	if err := rootCmd.Execute(); err != nil {
+		logrus.Error(trace.DebugReport(err))
+		os.Exit(255)
+	}
+}
 
+func root(ccmd *cobra.Command, args []string) error {
 	client, err := kubernetes.NewClient()
 	if err != nil {
-		exitWithError(err)
+		return trace.Wrap(err)
 	}
 
 	switch mode {
 	case constants.ModeDashboards:
 		err = runDashboardsWatcher(client)
 	case constants.ModeRollups:
-		err = runRollupsWatcher(client)
+		err = runRollupsWatcher(client, influxDBConfig)
 	case constants.ModeAlerts:
 		err = runAlertsWatcher(client)
 	default:
-		fmt.Printf("ERROR: unknown mode %q\n", mode)
-		os.Exit(255)
+		return trace.Errorf("ERROR: unknown mode %q\n", mode)
 	}
 
 	if err != nil {
-		exitWithError(err)
+		return trace.Wrap(err)
 	}
+
+	return nil
 }
 
-func exitWithError(err error) {
-	log.Error(trace.DebugReport(err))
-	fmt.Printf("ERROR: %v\n", err.Error())
-	os.Exit(255)
+// bindFlagEnv binds environment variables to command flags
+func bindFlagEnv(flagSet *flag.FlagSet) {
+	for env, flag := range envs {
```

Comment:
I would move `envs` inside this function - does not look like it's used elsewhere so it does not need to be package global